processes/process_sleepspindle_parameters/process_sleepspindle_parameters

```diff
@@ -0,0 +1,202 @@
+                                     'Nat Methods . 2014 Apr;11(4):385-92. doi: 10.1038/nmeth.2855."<BR><BR>'...
+                                     'Notes<BR>'...
+                                     'A) This process assumes that the data was already filtered between 11-16 Hz.<BR>'...
+                                     'B) This process will generate 4 files - duration, frequency, amplitude and symmetry.<BR>'...
+                                     'C) Each file x-axis = spindle number but the units will = Time(s). Cannot be changed.<BR>'...
+                                     'D) Follow the online tutorial which will take you to the GitHub README.md written by the<BR>'...
+                                     'author of this process to further understand how the process works.<BR><BR>'
+                                     ];
+    sProcess.options.info.Type    = 'label';
+    sProcess.options.info.Value   = [];
+end
+
+
+%% ===== FORMAT COMMENT =====
+function Comment = FormatComment(sProcess) %#ok<DEFNU>
+     Comment = sProcess.Comment;
+end
+
+
+%% ===== RUN =====
+function OutputFiles = Run(sProcess, sInputs) %#ok<DEFNU>
+    % Initialize returned list of files
+    OutputFiles = {};
+    Nepochs     = length(sInputs); % Numer of inputs or epochs
+    EpochN      = linspace (1,(Nepochs),(Nepochs)); % Changes the x-axis from time (s) to epoch number n. 
+    DataMat     = in_bst(sInputs(1).FileName, [], 0); % Read the first file in the list, to obtain channel size. 
+    epochSize   = size(DataMat.F);    
+    Nchannels   = epochSize(1); % Number of ALL channels (i.e. EEG, EOG, EMG and status)
+
+    % Generate matrices of zeros (Nchannels X Nepochs) to concatenate results
+    SS_Dur = zeros (Nchannels,Nepochs); % Spindle duration
+    CountP = zeros (Nchannels,Nepochs); % Count the number of peaks
+    PeakPo = zeros (Nchannels,Nepochs); % Positive peaks
+    PeakNe = zeros (Nchannels,Nepochs); % Negative peaks 
+    TimePP = zeros (Nchannels,Nepochs); % Time of max positive peak
+    TimeNP = zeros (Nchannels,Nepochs); % Time of max negative peak
+
+    % ===== LOAD THE DATA =====
+    for iNepochs = 1 : Nepochs
+       DataMat   = in_bst(sInputs(iNepochs).FileName, [], 0);
+       epochSize = size(DataMat.F);  
+       Ntime     = epochSize(2); % Number of time samples changes depending on sleep spindle
+       Data      = DataMat.F; % Actual data containing EEG, EOG and EMG data
+       Time      = DataMat.Time(end); % Time information of the recording
+       Fs        = (Ntime-1)/Time; % Calculation of Sampling frequency
+
+    % ===== PROCESS =====
+    % This is where the actual process of data manipulation and calculation takes place.
+       DataN        = Data*-1; % The negative version of the data
+       Data(end,:)  = (randi(100,1,Ntime))*10^-6; % End row which is BDF/status channel is a constant number so replaced this with randi to remove issues during findpeaks 
+       DataN(end,:) = (randi(100,1,Ntime))*10^-6;
```

Comment:
Instead of setting the last channel to noise, it will be better to run the FOR loop only for EEG channels, and check that there are not flat.


```
% Get channel file for each input file
sChannel = bst_get('ChannelForStudy', [sInputs(iNepochs).iStudy]);
% Load channel information
ChannelMat = in_bst_channel(ChannelFile, 'Channel');
% Indices for EEG channels only
eegIxs = strcmpi({ChannelMat.Channel.Type}, 'EEG');

% Current FOR loop per channel
for eegIx = 1 : length(eegIxs)
    i = eegIxs(eegIx);
    Data_Chan = Data(i,:); % Extracts individual channel info from the data matrix   
    % Check for flat channel
    if sum((Data_Chan - mean(Data_Chan)).^2) == 0
        continue;
    end
    % Compute sleep spindles ...
end
```

Reply:
Note that the detection of flat channels could be optimized. I added in the FOR loop to make clearer

Reply:
Thanks for the input @rcassani.
I've made some adjustments but keep getting the following error message which I cannot resolve. What are your thoughts? 
![image](https://github.com/brainstorm-tools/bst-users/assets/134780775/aac44457-a01f-4419-a9c9-6133f1febe72)

```
    % ===== LOAD THE DATA =====
    for iNepochs = 1 : Nepochs
       DataMat   = in_bst(sInputs(iNepochs).FileName, [], 0);
       epochSize = size(DataMat.F);  
       Ntime     = epochSize(2); % Number of time samples changes depending on sleep spindle
       Data      = DataMat.F; % Actual data containing EEG, EOG and EMG data
       Time      = DataMat.Time(end) - DataMat.Time(1); % Time information of the recording
       Fs        = 1 ./ (DataMat.Time(2) - DataMat.Time(1)); % Calculation of Sampling frequency

       % Get channel file for each input file
       sChannel = bst_get('ChannelForStudy', [sInputs(iNepochs).iStudy]);
       % Load channel information
       ChannelMat = in_bst_channel(sChannel, 'Channel');
       % Indices for EEG channels only
       eegIxs = strcmpi({ChannelMat.Channel.Type}, 'EEG');

    % ===== PROCESS =====
    % This is where the actual process of data manipulation and calculation takes place.
       DataN        = Data*-1; % The negative version of the data
       %Data(end,:)  = (randi(100,1,Ntime))*10^-6; % End row which is BDF/status channel is a constant number so replaced this with randi to remove issues during findpeaks 
       %DataN(end,:) = (randi(100,1,Ntime))*10^-6;

            %for i = 1 : Nchannels
            for eegIx = 1 : length (eegIxs)
                i = eegIxs(eegIx);
                Data_Chan = Data(i,:); % Extracts individual channel info from the data matrix 
                if sum((Data_Chan - mean(Data_Chan)).^2) == 0 % Check for flat channel
                 continue;
                end
% compute spindles...
```